lib/Serialization/ScanningLoaders.cpp

```diff
@@ -67,19 +67,27 @@ std::error_code SwiftModuleScanner::findModuleFilesInDirectory(
   }
   assert(fs.exists(InPath));
 
-  // Use the private interface file if exits.
-  auto PrivateInPath =
-      BaseName.getName(file_types::TY_PrivateSwiftModuleInterfaceFile);
-  if (fs.exists(PrivateInPath)) {
-    InPath = PrivateInPath;
+  // Use package.swiftinterface if it exists and its package-name applies to
+  // the importer module.
+  auto PkgInPath = getPackageInterfacePathIfInSamePackage(fs, Ctx);
+  if (fs.exists(PkgInPath)) {
+    InPath = PkgInPath;
+  }
+
+  // If not in package, use the private interface file if exits.
+  if (fs.exists(InPath)) {
```

Comment:
Shouldn't this be an `else if` to avoid overriding the package swiftinterface if it was picked?

Reply:
Oh, I meant the negation of that; will fix it.